syntax = "proto3";

package kvstore;

// The gRPC service definition for lifecycle management of KV nodes
service KVNodeLifecycle {
  rpc Shutdown(ShutdownRequest) returns (ShutdownResponse) {}
  rpc Restart(RestartRequest) returns (RestartResponse) {}
  rpc CheckNodeHealth(HealthCheckRequest) returns (HealthCheckResponse) {}
  rpc SetPrimary(SetPrimaryRequest) returns (SetPrimaryResponse) {}
  rpc ActiveNodes(ActiveNodesRequest) returns (ActiveNodesResponse) {}
}

message ShutdownRequest {}
message ShutdownResponse {}

message RestartRequest {}
message RestartResponse {}

message HealthCheckRequest {}
message HealthCheckResponse {
  bool isHealthy = 1;
  string recovery_status = 2;
}

message SetPrimaryRequest {
  string primaryAddress = 1;
}
message SetPrimaryResponse {}

message ActiveNodesRequest {
  repeated string nodeAddresses = 1;
}
message ActiveNodesResponse {}
